views/js/controller/runner/runner.js

```diff
@@ -195,79 +165,50 @@ define([
-             */
-            var loadProviders = function loadProviders(providers){
-
-                return providerLoaderFactory()
-                    .addList(_.filter(providers, {category: 'runner'}))
-                    .load(context.bundle);
-            };
-
-            /**
-             * Load the configured proxy provider
-             * @returns {Promise} resolves with the name of the proxy provider
-             */
-            var loadProxy = function loadProxy(){
-                return proxyLoader();
-            };
-
-            var moduleConfig = module.config();
+            const moduleConfig = module.config();
 
             loadingBar.start();
 
-            // verify required options
-            if( ! _.every(requiredOptions, hasOption)) {
-                return onError(new TypeError(__('Missing required option %s', name)));
+            // verify required config
+            if ( ! requiredOptions.every(hasOption) ) {
+                return onError(new TypeError(__('Missing required configuration option %s', name)));
             }
 
             // dispatch any extra registered routes
             if (moduleConfig && _.isArray(moduleConfig.extraRoutes) && moduleConfig.extraRoutes.length) {
                 router.dispatch(moduleConfig.extraRoutes);
             }
 
+            //for the qti provider to be selected here
+            config.provider = Object.assign( config.provider || {}, { runner: 'qti' });
+
             //load the plugins and the proxy provider
-            Promise
-                .all([
-                    loadPlugins(runnerOptions.plugins),
-                    loadProxy()
-                ])
+            providerLoader(config.providers, context.bundle)
                 .then(function (results) {
 
-                    var plugins = results[0];
-                    var proxyProviderName = results[1];
+                    const testRunnerConfig = _.omit(config, ['providers']);
+                    testRunnerConfig.renderTo = $container;
 
-                    var config = _.omit(runnerOptions, ['plugins', 'providers']);
-                    config.proxyProvider = proxyProviderName;
-                    config.renderTo      = $container;
+                    if (results.proxy && typeof results.proxy.getAvailableProviders === 'function') {
+                        const loadedProxies = results.proxy.getAvailableProviders();
+                        testRunnerConfig.provider.proxy = loadedProxies[0];
+                    }
 
-                    logger.debug({ config: config, plugins: plugins}, 'Start test runner');
+                    logger.debug({
+                        config: testRunnerConfig,
+                        providers : config.providers
+                    }, 'Start test runner');
 
                     //instantiate the QtiTestRunner
-                    runner('qti', plugins, config)
+                    runner('qti', results.plugins, testRunnerConfig)
```

Comment:
we are in taoQtiTest, so that was my guess to keep using the QTI provider (this was the previous behavior) even if we are now loading the providers dynamically

Reply:
I've changed it to be prepared if in the future we want to use another provider (but for me we will have to use the runnerComponent instead, which use directly the loaded factory)

Reply:
ok, sounds good